src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -127,8 +127,17 @@ public unsafe sealed override bool IsReflectionBlocked(RuntimeTypeHandle runtime
             runtimeTypeHandle = GetTypeDefinition(runtimeTypeHandle);
 
             var moduleHandle = RuntimeAugments.GetModuleFromTypeHandle(runtimeTypeHandle);
-            NativeFormatModuleInfo module = ModuleList.Instance.GetModuleInfoByHandle(moduleHandle);
+            ModuleInfo untypedModuleInfo;
+            NativeFormatModuleInfo module = null;
 
+            //make sure the module is actually NativeFormatModuleInfo, in platforms where
+            //real reflection isnt enabled the module will not be a NativeFormatModuleInfo
```

Comment:
> in platforms where real reflection isnt enabled 

It would be more accurate to say "for modules without reflection data". There can be multiple modules loaded, and reflection can be disabled only for some of them.